Feature/apollo typescript setup

JonasBorgesLM opened the pull request:
Step 4/5 - Typescript and Apollo Server

JonasBorgesLM committed:
refactor: :recycle: adicionando apollo e typescript

O servidor foi refeito utilizando o apollo e typescript

JonasBorgesLM committed:
docs: :memo: update - README.md

Atualizando a documentacao (README.md) para nova versao do servidor

matheusflds commented on README.md:
```diff
@@ -6,18 +6,20 @@ Este projeto é um sistema de gerenciamento de usuários que permite **criar, le
 ### Pré-requisitos
 
-Certifique-se de ter o Node.js instalado. Recomenda-se a versão 18.20.4
+- Certifique-se de ter o Node.js instalado. Recomenda-se a versão 18.20.4
```
não vai fazer muita diferença no escopo do onboard, creio, mas pode ser uma boa já utilizar a versão LTS (20.16.0, se não me engano)

matheusflds commented on src/schema/helloSchema.ts:
a título de padronização, costumamos usar `kebab-case` para nomear os arquivos (`helloSchema.ts -> hello-schema.ts`)

matheusflds commented on tsconfig.json:
```diff
@@ -0,0 +1,14 @@
+  "compilerOptions": {
+    "target": "ES2020",
+    "module": "commonjs",
+    "outDir": "./dist",
```
precisa adicionar essa pasta dist no `.gitignore` também, do contrário vai commitar (sem necessidade) o código transpilado pro js também

matheusflds commented on tsconfig.json:
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "target": "ES2020",
+    "module": "commonjs",
```
aqui eu recomendaria utilizar o esnext, provavelmente vai acaretar nuns ajustes nas proximas libs que vc instalar no botinho, mas é por uma boa causa haha
Ainda não tenho certeza do projeto que vocês vão ser alocados, mas um de vocês vai pra MD, que já ta no esnext. Assim, acho uma boa vocês dois já fazerem o botinho utilizando o esnext, porque é muito mais fácil "voltar" do esnext pro commonjs do que o oposto.

c.c. @bialima2001 (referente a parte de utilizar o module esnext

JonasBorgesLM committed:
refactor: :recycle: melhorias levantadas pelo CR

Melhorias levantadas pelo Code Review. Englobando versao do node 20.16.0, esnext, nodemon e padronizacao do nome dos arquivos (hello-schema)

matheusflds reviewed (approved):
só colocar as quebra de linha no nodemon.json e package.json e sucesso 🚀
